backend/kernelCI_app/views/treeCommitsHistory.py

```diff
@@ -1,155 +1,285 @@
+        self.commit_hashes[commit_hash]['tests_count'][label] += 1
+
+    def _pass_in_global_filters(self, row: Dict) -> bool:
+        hardware_compatibles = [UNKNOWN_STRING]
+        architecture = UNKNOWN_STRING
+        compiler = UNKNOWN_STRING
+        config_name = UNKNOWN_STRING
+
+        if row['hardware_compatibles'] is not None:
+            hardware_compatibles = row['hardware_compatibles']
+        if row['architecture'] is not None:
+            architecture = row['architecture']
+        if row['compiler'] is not None:
+            compiler = row['compiler']
+        if row['config_name'] is not None:
+            config_name = row['config_name']
+
+        if (
+            (
+                len(self.filterHardware) > 0
+                and (not self.filterHardware.intersection(hardware_compatibles))
+            )
+            or (
+                len(self.filterArchitecture) > 0
+                and (architecture not in self.filterArchitecture)
+            )
+            or (
+                len(self.filterTreeDetailsCompiler) > 0
+                and (compiler not in self.filterTreeDetailsCompiler)
+            )
+            or (
+                len(self.filterTreeDetailsConfigs) > 0
+                and (config_name not in self.filterTreeDetailsConfigs)
+            )
+        ):
+            return False
+
+        return True
+
+    def _process_tests(self, row: Dict, commit_hash: str) -> None:
+        if row['test_id'] is not None and row['test_id'] not in self.processed_tests:
+            self.processed_tests.add(row['test_id'])
+            is_boot = row['test_path'] is not None and row['test_path'].startswith('boot')
+
+            if is_boot:
+                self._process_boots_count(
+                    row['test_status'] or "NULL",
+                    commit_hash,
+                    row['test_duration'],
+                    row['test_path']
+                )
+            else:
+                self._process_nonboots_count(
+                    row['test_status'] or "NULL",
+                    commit_hash,
+                    row['test_duration'],
+                    row['test_path']
+                )
 
-            if field != "environment_compatible":
-                self.filters_options[table]['filters'].append(clause)
+    def _process_builds(self, row: Dict, commit_hash: str) -> None:
+        if row["build_id"] is not None and row['build_id'] not in self.processed_builds:
+            self.processed_builds.add(row['build_id'])
+            self._process_builds_count(row["build_valid"], row['build_duration'], commit_hash)
 
-            if field in ["config_name", "architecture", "compiler", "environment_compatible"]:
-                self.filters_options['test']['filters'].append(clause)
-                self.filters_options['boot']['filters'].append(clause)
+    def _process_rows(self, rows: Dict) -> None:
+        rowsSanitize = self.sanitize_rows(rows)
```

Comment:
use snake_case for variable names, `sanitized_rows` would also be a better name, in my opinion